boards/arm/qemu_cortex_r5/board.cmake

```diff
@@ -5,12 +5,15 @@
 set(EMU_PLATFORM qemu)
 set(QEMU_ARCH xilinx-aarch64)
 
+if(CONFIG_QEMU_ICOUNT)
+  set(ICOUNT_ARG -icount shift=3,align=off,sleep=off -rtc clock=vm)
+endif()
 set(QEMU_CPU_TYPE_${ARCH} cortex-r5)
 set(QEMU_FLAGS_${ARCH}
   -nographic
   -machine arm-generic-fdt
-  -icount shift=3,align=off,sleep=off -rtc clock=vm
   -dtb ${ZEPHYR_BASE}/boards/${ARCH}/${BOARD}/fdt-single_arch-zcu102-arm.dtb
+  ${ICOUNT_ARG}
   )
```

Comment:
No need to give `qemu_cortex_r5` a special treatment, we should be able to just append here as well.

Reply:
@stephanosio thanks, updated.